debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/YbExporterConsumer.java

```diff
@@ -80,6 +93,97 @@ void connect() throws URISyntaxException {
         flusherThread = new Thread(this::flush);
         flusherThread.setDaemon(true);
         flusherThread.start();
+
+        // Register shutdown hook to release lock on exit
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            releaseLockFile();
+        }));
+    }
+
+    /**
+     * Reads the lock file and checks if the PID is still running.
+     * If the PID is not running, it deletes the lock file.
+     * If the PID is running, it throws an IllegalStateException.
+     */
+    private void readLockFileAndCheckPid() {
+        try {
+            String lockContent = Files.readString(lockFile.toPath());
+            String[] lines = lockContent.split("\n");
+            if (lines == null || lines.length == 0) {
+                // If the lock file is empty, error out
+                String msg = String.format("Lock file {} is empty.", lockFile.getAbsolutePath());
+                throw new IllegalStateException(msg);
+            }
+            String pid = lines[0].trim();
+            // Check if the PID is a valid number
+            LOGGER.info("Lock file {} exists with PID: {}", lockFile.getAbsolutePath(), pid);
+            //Parse PID and check if the process is running
+            try {
+                long pidLong = Long.parseLong(pid);
+                // Check if the process with this PID is running
+                if (ProcessHandle.of(pidLong).isPresent()) {
+                    // Process is running, error out
+                    String msg = String.format("Lock file %s already exists and process with PID %s is running. Another process may be running for this dataDir.", lockFile.getAbsolutePath(), pid);
+                    LOGGER.error(msg);
+                    throw new IllegalStateException(msg);
+                } else {
+                    // Process is not running, we can safely delete the lock file
```

Comment:
Hmm, I'm slightly skeptical of this. I would have just thrown an error, and the user can always manually delete the lock file. Do we delete it like this in the golang layer?

Reply:
OKay let's go ahead then 👍